galaxy/templates/deployment-job.yaml

```diff
@@ -137,7 +137,7 @@ spec:
               subPath: probedb.py
             {{- range $key, $entry := $.Values.extraFileMappings -}}
             {{- if $entry.applyToJob }}
-            - name: extra-files-{{ include "galaxy.getFilenameFromPath" $key | replace "." "-" }}
+            - name: {{ include "galaxy.makeDnsCompliant" (printf "%extra-%s-%s" (include "galaxy.getFilenameFromPath" $key) ($key | sha256sum)) }}
```

Comment:
Since this code segment for figuring out the filename has been repeated and is also a bit difficult to understand at first glance, can we move it to a template function, and then document what's going on in the template? e.g. Why is the sha256sum needed here etc? (presumably for uniqueness?)

something like: `get_extra_files_entry_name()` and `get_extra_files_secret_name()` or something?

Also, this particular line seems to be incorrect - should be`%s-%s` instead of `%extra-%s-%s'?

Reply:
That is definitely a typo, was supposed to be just `extra`. The `sha256sum` is just for uniqueness (for the case where it's the same filename but different paths). Sure i'll put things in a function